### PR TITLE
Merged master into ddale's PR

### LINKED ISSUE #266: conda-build failing to report git describe info to templated yaml file
I'm working on a project that keeps its conda recipe in PROJROOT/conda.recipe. In `meta.yaml` file, I have:

```
package:
  name: hexrd
  version: {{ environ['GIT_DESCRIBE_TAG'][1:]+environ.get('GIT_DESCRIBE_NUMBER', '') }}

source:
  path: ../

```

When I try to build it, I get:

```
$ conda build conda.recipe/
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda-build/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 5, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python2.7/site-packages/conda_build/main_build.py", line 110, in main
    args_func(args, p)
  File "/opt/conda/lib/python2.7/site-packages/conda_build/main_build.py", line 312, in args_func
    args.func(args, p)
  File "/opt/conda/lib/python2.7/site-packages/conda_build/main_build.py", line 244, in execute
    m = MetaData(recipe_dir)
  File "/opt/conda/lib/python2.7/site-packages/conda_build/metadata.py", line 250, in __init__
    self.parse_again()
  File "/opt/conda/lib/python2.7/site-packages/conda_build/metadata.py", line 258, in parse_again
    self.meta = parse(get_contents(self.meta_path))
  File "/opt/conda/lib/python2.7/site-packages/conda_build/metadata.py", line 234, in get_contents
    contents = template.render(environment=env)
  File "/opt/conda/lib/python2.7/site-packages/jinja2/environment.py", line 969, in render
    return self.environment.handle_exception(exc_info, True)
  File "/opt/conda/lib/python2.7/site-packages/jinja2/environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/darren/Projects/hexrd/conda.recipe/meta.yaml", line 4, in top-level template code
    version: {{ environ['GIT_DESCRIBE_TAG'][1:]+environ.get('GIT_DESCRIBE_NUMBER', '') }}
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'GIT_DESCRIBE_TAG'
```

Inspecting `environ.py:get_dict`, which calls `os.path.join(d['SRC_DIR'], '.git')`, I see the value of that path is `/home/darren/conda-bld/work/PyQt-x11-gpl-4.11.2/.git`, which is the previous conda package I attempted to build, but the build failed. I deleted my old `conda-bld/work`, and when I tried again to build the hexrd recipe, conda-build looked for `/home/darren/conda-bld/work/.git`. It appears that conda-build is not unpacking the source into conda-bld/work before attempting to get environment information that requires the source.

```
$ conda info
Current conda install:

             platform : linux-64
        conda version : 3.7.3
  conda-build version : 1.9.1
       python version : 2.7.8.final.0
     requests version : 2.5.0
     root environment : /opt/conda  (read only)
  default environment : /opt/conda
     envs directories : /home/darren/.conda/envs
                        /home/darren/envs
                        /opt/conda/envs
        package cache : /home/darren/.conda/envs/.pkgs
                        /home/darren/envs/.pkgs
                        /opt/conda/pkgs
         channel URLs : http://conda.binstar.org/CHESS/linux-64/
                        http://conda.binstar.org/dsdale24/linux-64/
                        http://conda.binstar.org/praxes/linux-64/
                        http://repo.continuum.io/pkgs/free/linux-64/
                        http://repo.continuum.io/pkgs/pro/linux-64/
          config file : /home/darren/.condarc
    is foreign system : False
```

### PR DESCRIPTION
**Pull request by tswicegood**
Just a merge of master against #268 that fixes #266.

### CODE REVIEW

**Comment by tswicegood** [comment]
All good!  Merged.

File: conda_build/build.py [hunk 1]
```diff
@@ -322,7 +322,7 @@ def build(m, get_src=True, verbose=True, post=None):
 
         # Display the name only
         # Version number could be missing due to dependency on source info.
-        print("BUILD START:", m.dist())
+        print("BUILD START:", m.name())
```
Comment (asmeurer) [inline conda_build/build.py]: @ddale @tswicegood why was this changed?

Comment (tswicegood) [inline conda_build/build.py]: Don't know -- that's what @ddale had done.  I believe it'll avoid giving bad information, however.  `m.name()` doesn't change, but the output of `m.dist()` does if git is involved.

Comment (ddale) [inline conda_build/build.py]: What @tswicegood said. Also, the comment on the lines above the change address the issue.
